src/metabase/util/jvm.clj

```diff
@@ -299,3 +299,18 @@
   "Run `body` in a `future` and throw an exception if it fails to complete after `timeout-ms`."
   [timeout-ms & body]
   `(do-with-timeout ~timeout-ms (fn [] ~@body)))
+
+(defn wait-until
+  "Wait until `pred-fn` returns truthy, or until `max-retry` times, with `interval-ms` between retries."
+  [pred-fn & {:keys [max-retry interval-ms]
+              :or   {max-retry 10
+                     interval-ms 100} :as opts}]
+
+  (loop [retry-count 0]
+    (when-not (pred-fn)
+      (if (>= retry-count max-retry)
+        (throw (ex-info "Timeout" {:opts    opts
+                                   :pred-fn pred-fn}))
+        (do
+         (Thread/sleep interval-ms)
+         (recur (inc retry-count)))))))
```

Comment:
I created a function very similar to this the other day. Do you mind moving it to this namespce and reusing it for your test?
https://github.com/metabase/metabase/blob/936c185a874ff25a7ecc4aa69947e3902a058cf4/src/metabase/integrations/slack.clj#L320-L334

Reply:
nice helper. updated 58cb3f80ac